src/hooks/api/userQuery.ts

```diff
@@ -11,3 +14,36 @@ export const useGetUserQuery = () => {
   const client = useBoclipsClient();
   return useQuery(['user'], async () => doGetUser(client));
 };
+
+export const doCreateNewUser = (
+  request: CreateUserRequest,
+  client: BoclipsClient,
+) => {
+  return client.users.createUser(request);
+};
+
+export const useAddNewUser = () => {
+  const client = useBoclipsClient();
+
+  return useMutation(
+    (userRequest: CreateUserRequest) => doCreateNewUser(userRequest, client),
+    {
+      onSuccess: (userRequest) => {
+        displayNotification(
```

Comment:
I'm not 100% sure on best practices around queries, but what do you think about decoupling query logic from frontend logic? I.e. instead of displaying the notification directly from here, pass success/failure callbacks to the mutation in the modal component?